### DISCUSSION TITLE
The memory usage does not get released with ImageMagick service

### DISCUSSION THREAD

**Discussion by nanfeng1999** [ImageMagick/ImageMagick#6778] [Help]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

CentOS release 6.3 (Final)

### Description

The memory usage of the ImageMagick service does not get released, but it has an upper limit, and the memory usage does not increase beyond a certain threshold. Is this caused by ImageMagick?

**Comment by dlemstra** [comment]
What is "the ImageMagick service"?

**Reply by nanfeng1999** [reply]
ImageMagick is invoked as a library within our application.

**Reply by dlemstra** [reply]
I suspect that the application is using the library incorrect and that it is not releasing the memory.

**Reply by nanfeng1999** [reply]
So how to correctly release memory? Does ImageMagick provide corresponding interface functions?

**Reply by dlemstra** [reply]
That all depends on how you are using the library so I cannot give a clear answer to that question.

**Reply by urban-warrior** [reply]
ImageMagick employs constructors and destructors to manage image containers, each of which contains a pointer to an image pixel cache. To enhance efficiency, it's possible for multiple image containers to reference the same pixel cache. When using your service, it's essential to ensure that every image created is promptly destroyed when it's no longer required. By adhering to this advice, your service can become self-sustaining, as it will automatically reclaim and reuse resources as per the prevailing demand.
